lib/cocoapods/validator.rb

```diff
@@ -361,13 +368,53 @@ def download_pod
       @file_accessor = @installer.pod_targets.flat_map(&:file_accessors).find { |fa| fa.spec.name == consumer.spec.name }
     end
 
+    def create_app_project
+      app_project = Xcodeproj::Project.new(validation_dir + 'App.xcodeproj')
+      deployment_target = spec.subspec_by_name(subspec_name).deployment_target(consumer.platform_name)
+      app_project.new_target(:application, 'App', consumer.platform_name, deployment_target)
+      app_project.save
+      app_project.recreate_user_schemes
+      Xcodeproj::XCScheme.share_scheme(app_project.path, 'App')
+    end
+
+    def add_app_project_import
+      app_project = Xcodeproj::Project.open(validation_dir + 'App.xcodeproj')
+      pod_target = @installer.pod_targets.find { |pt| pt.pod_name == spec.root.name }
+      language = pod_target.uses_swift? ? :swift : :objc
+
+      if language == :swift
+        source_file = validation_dir.+('App/main.swift')
+        source_file.parent.mkpath
+        import_statement = use_frameworks ? "import #{pod_target.product_module_name}\n" : ''
+        source_file.open('w') { |f| f << import_statement }
+      else
+        source_file = validation_dir.+('App/main.m')
+        source_file.parent.mkpath
+        import_statement = if use_frameworks
+          "@import #{pod_target.product_module_name};\n"
+        else
+          header_name = "#{pod_target.product_module_name}/#{pod_target.product_module_name}.h"
+          if pod_target.sandbox.public_headers.root.+(header_name).file?
```

Comment:
this logic needs to be extracted somewhere, probably into `PodTarget` so we can also skip creating an umbrella header if it exists?